platform-tooling-support-tests/src/archUnit/java/platform/tooling/support/tests/ArchUnitTests.java

```diff
@@ -76,6 +80,32 @@ class ArchUnitTests {
 			.should(haveContainerAnnotationWithSameRetentionPolicy()) //
 			.andShould(haveContainerAnnotationWithSameTargetTypes());
 
+	@ArchTest
+	void packagesShouldBeNullMarked(JavaClasses classes) {
+		var exclusions = Stream.of( //
+			"..shadow..", //
+			"org.junit.jupiter.api..", //
+			"org.junit.jupiter.engine..", //
+			"org.junit.jupiter.migrationsupport..", //
+			"org.junit.jupiter.params..", //
+			"org.junit.platform.launcher.." //
+		).map(PackageMatcher::of).toList();
+
+		var subpackages = Stream.of("org.junit.platform", "org.junit.jupiter", "org.junit.vintage") //
+				.map(classes::getPackage) //
+				.flatMap(rootPackage -> rootPackage.getSubpackagesInTree().stream()) //
+				.filter(pkg -> exclusions.stream().noneMatch(it -> it.matches(pkg.getName()))) //
+				.filter(pkg -> !pkg.getClasses().isEmpty()) //
+				.toList();
+		assertThat(subpackages).isNotEmpty();
+
+		var violations = subpackages.stream() //
+				.filter(pkg -> !pkg.isAnnotatedWith(NullMarked.class)) //
+				.map(JavaPackage::getName) //
+				.sorted();
+		assertThat(violations).describedAs("The following packages are missing the @NullMarked annotation").isEmpty();
+	}
```

Comment:
This check will help us to avoid forgetting to add the annotation on new packages.